backend/src/api/model/playlist/mutations.rs

```diff
@@ -74,20 +76,37 @@ impl AuthorizedPlaylist {
         id: Id,
         title: Option<String>,
         description: Option<String>,
-        entries: Option<Vec<String>>,
+        entries: Option<Vec<Id>>,
         acl: Option<Vec<AclInputEntry>>,
         context: &Context,
     ) -> ApiResult<Self> {
         // `load_for_mutation` handles authorization.
         let playlist = Playlist::load_for_mutation(id, context).await?;
 
+        let entry_ids = if let Some(entries) = entries {
+            Some(futures::stream::iter(entries)
+                .then(|id| async move {
+                    let maybe_event = AuthorizedEvent::load_by_id(id, context).await?;
+                    let event = maybe_event.ok_or_else(|| err::invalid_input!(
+                        key = "event.not-found",
+                        "unknown event"
+                    ))?;
+                    let event = event.into_result()?;
+                    Ok::<_, ApiError>(event.opencast_id)
+                })
+                .try_collect::<Vec<_>>()
+                .await?)
```

Comment:
Mhhh why this stream thingy? Feel like this should be doable with a single query that is then also easier for the Rust code? Of course, yeah, one needs to keep the order, but should be possible somehow. Does the current code keep the order?

Reply:
Yeah that was a dumb idea. No need to get fancy here and as you noted, doing it async wouldn't preserve the order.